libethcore/Farm.h

```diff
@@ -83,6 +83,10 @@ class GenericFarm: public GenericFarmFace<PoW>
 	 */
 	bool start(std::string const& _sealer)
 	{
+		// call empty setWork to make sure that all workers have stoppedWorking()
+		for (auto& miner: m_miners)
+			miner->setWork();
+
```

Comment:
this is not thread-safe. `m_miners` could change from another thread.

`m_miners.clear()` (happens later) has the same effect but is actually thead-safe.